### ISSUE TITLE
[v2] Auto-generate configuration documentation

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger#6186]

**Comment by mahadzaryab1** [comment]
@yurishkuro how should we get started on this?

**Comment by yurishkuro** [comment]
Can't, blocked by upstream issues

**Comment by llvm-x86** [comment]
@yurishkuro what about now? How may I start contributing?

**Comment by AdityaKrSingh26** [comment]
Hey @yurishkuro @jkowall 

I'm looking at this issue and wanted to discuss my approach before starting so that i can get a better understanding of this. 

Here's my understanding of the problem:

  Current State:
  - Jaeger-v2 config documentation is manual (YAML examples only)
  - Config structs are spread across 6 extensions (jaegerquery, jaegerstorage, remotesampling, etc.) plus 40+ OTEL components
  - When code changes, docs can get out of sync

  Goal:
  Build a tool that auto-generates JSON Schema docs from Go structs, including:
  - Field types (from struct definitions)
  - Descriptions (from Go comments)
  - Default values (from struct instances)
  - Required fields (from struct tags)

  Does this match your understanding?

**Comment by AdityaKrSingh26** [comment]
Should we start with just the 6 Jaeger extensions (jaegerquery, jaegerstorage, etc.) or include all 40+ OTEL components too? My thought: start small for faster MVP?

If I can't parse OTEL/third-party source code, is it okay to skip comments for those types?

what are your thoughts on this ? @yurishkuro @jkowall

**Comment by yurishkuro** [comment]
> Build a tool that auto-generates JSON Schema docs from Go structs

This isn't the end goal, the goal is to build documentation on https://www.jaegertracing.io/ for all configuration options / structs. In practice generating JSON Schema is likely the correct implementation detail as there are many tools / libraries that can take JSON Schema and product documentation in Markdown.

> Should we start with just the 6 Jaeger extensions

Yes. We already refer to upstream OTEL docs for their extensions so as they improve those we will benefit. But our internal extension configs are undocumented right now.

**Comment by AdityaKrSingh26** [comment]
> > Build a tool that auto-generates JSON Schema docs from Go structs
> 
> This isn't the end goal, the goal is to build documentation on https://www.jaegertracing.io/ for all configuration options / structs. In practice generating JSON Schema is likely the correct implementation detail as there are many tools / libraries that can take JSON Schema and product documentation in Markdown.
> 
> > Should we start with just the 6 Jaeger extensions
> 
> Yes. We already refer to upstream OTEL docs for their extensions so as they improve those we will benefit. But our internal extension configs are undocumented right now.

Okay ... got it 🫡

**Comment by SoumyaRaikwar** [comment]
@yurishkuro I've implemented schemagen for internal extensions (#7947 ). OTel cmd/configschema/docsgen is deprecated/removed

approach for doc generation work 
 - Use AST parsing like #6628 for Jaeger-native docgen (structs → Markdown/YAML).
- Add `make docs` with `jaeger print-config` for merged output.

**Comment by Audio431** [comment]
Trying to piece together where this sits now across the related threads:

- **Direction (issue body):** schema-first — schema → Go + validation + docs
- **[#7947](https://github.com/jaegertracing/jaeger/pull/7947)** (merged, Feb 2026): schemagen (Go → schema) run against internal extensions
- **[#8089](https://github.com/jaegertracing/jaeger/pull/8089)** (stalled Feb 27): attempted JSON schema generation, concluded custom JSON Schema → Go generation in Jaeger alone is an anti-pattern due to OTEL type coupling
- **[#6628](https://github.com/jaegertracing/jaeger/issues/6628)** (AST workaround): converged on aligning with upstream schema-first over reflection
- **Upstream:** RFC [open-telemetry/opentelemetry-collector#14433](https://github.com/open-telemetry/opentelemetry-collector/pull/14433) merged Jan 24 establishing schema-first roadmap (YAML as source of truth → Go + JSON schemas + docs)

Both Jaeger threads have converged on aligning with upstream rather than building parallel infrastructure. Is the current state essentially waiting for the upstream schema → Go path to become usable?

**Comment by yurishkuro** [comment]
Yes, and I don't think https://github.com/open-telemetry/opentelemetry-collector/pull/14433 is the last word, there has been progress since January.

**Comment by Audio431** [comment]
@yurishkuro Thanks for the clarification.

**Comment by amh1k** [comment]
is the issue still open?

**Comment by Retr0-XD** [comment]
can I work on this issue? , is the issue still open ?
